pkg/controller/atlas/connection.go

```diff
@@ -0,0 +1,55 @@
+package atlas
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Connection encapsulates Atlas connectivity information that is necessary to perform API requests
+type Connection struct {
+	OrgID      string
+	PublicKey  string
+	PrivateKey string
+}
+
+// ReadConnection reads Atlas API connection parameters from AtlasProject Secret or from the default Operator one if the
+// former is not specified
+func ReadConnection(kubeClient client.Client, operatorName string, projectOverrideSecretRef *client.ObjectKey, log *zap.SugaredLogger) (Connection, error) {
+	if projectOverrideSecretRef != nil {
+		// TODO is it possible that part of connection (like orgID is still in the Operator level secret and needs to get merged?)
+		log.Infof("Reading Atlas API credentials from the AtlasProject Secret %s", projectOverrideSecretRef)
+		return readAtlasConnectionFromSecret(kubeClient, *projectOverrideSecretRef)
+	}
+	// TODO check the default "Operator level" Secret
+	// return readAtlasConnectionFromSecret(operatorName + "-connection")
+	return Connection{}, errors.New("The API keys are not configured")
+}
+
+func readAtlasConnectionFromSecret(kubeClient client.Client, secretRef client.ObjectKey) (Connection, error) {
+	secret := &corev1.Secret{}
+	if err := kubeClient.Get(context.Background(), secretRef, secret); err != nil {
+		return Connection{}, err
+	}
+	secretData := make(map[string]string)
+	for k, v := range secret.Data {
+		secretData[k] = string(v)
+	}
+	if _, ok := secretData["orgId"]; !ok {
+		return Connection{}, fmt.Errorf("Missing field with key 'orgId' in the secret %v", secretRef)
+	}
+	if _, ok := secretData["publicApiKey"]; !ok {
+		return Connection{}, fmt.Errorf("Missing field with key 'publicApiKey' in the secret %v", secretRef)
+	}
+	if _, ok := secretData["privateApiKey"]; !ok {
+		return Connection{}, fmt.Errorf("Missing field with key 'privateApiKey' in the secret %v", secretRef)
+	}
```

Comment:
[q] given you are wrapping `secretRef` could this error leak secrets? ie orgID is not set but api keys are and we end up logging this error somewhere?

Reply:
also optional/future improvement, I find tools that error on a missing field just to error again on another one, it would be nice to check all required fields and returning once with all the missing info instead of me having to run this again to find a new piece of missing information

Reply:
re 1: don't think we'll leak anything as only the secret name is printed, not its content
re 2: I agree with you, there is some duplication here as well, can be done by using an array and throwing the error once 👍